### PR TITLE
[GEOS-10495] Request Logger Memory Buffer Limits

### PR DESCRIPTION
**Pull request by turingtestfail**
[GEOS-10495](https://osgeo-org.atlassian.net/browse/GEOS-10495)

Added unit tests

documentation

cleanup

cleanup 2

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Review by aaime — CHANGES_REQUESTED** [review]
Looks good, see feedback below.

File: doc/en/user/source/configuration/globalsettings.rst [hunk 1]
```diff
@@ -216,17 +216,6 @@ If you are running GeoServer in a large J2EE container, you might not want your
 
 This setting can be overriden by system property, see :ref:`logging` for details (this setting removes Console ``stdout`` appender).
 
-.. _config_globalsettings_log_buffer:
```
[the remark this answers is not in the document]
Comment (turingtestfail) [reply doc/en/user/source/configuration/globalsettings.rst:219]: fixed

File: doc/en/user/source/configuration/globalsettings.rst [hunk 2]
```diff
@@ -238,7 +227,10 @@ These settings enable the logging of the requested URL, and optionally request h
 
 * :guilabel:`Log Request Bodies`: Select to enable logging of request body information when accepting (``POST``,``PUT``) requests. Text content will be logged, or the number of bytes for binary content, based on the setting :ref:`config_globalsettings_log_buffer` setting above.
 
-* :guilabel:`Log Request Bodies`: Select to enable logging of request header information.
+* :guilabel:`Number of characters to log for incoming POST requests`: In more verbose logging levels, GeoServer will log the body of XML (and other text formats) recieved by ``POST`` and ``PUT`` requests. It will only log the initial part of the request though, since it has to store (buffer) everything that gets logged for use in the parts of GeoServer that use it normally. This setting sets the size of this buffer, in characters.  A setting of **0** will disable the log buffer.
```
[the remark this answers is not in the document]
Comment (turingtestfail) [reply doc/en/user/source/configuration/globalsettings.rst]: fixed

File: src/main/src/main/java/org/geoserver/filters/LoggingFilter.java [hunk 3]
```diff
@@ -92,6 +100,12 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
             logHeaders =
                     (metadataMap.containsKey(LOG_HEADERS_ENABLED)
                             && metadataMap.get(LOG_HEADERS_ENABLED, Boolean.class));
+            requestLogBufferSize =
+                    geoServer.getGlobal().getXmlPostRequestLogBufferSize() != null
+                            ? geoServer.getGlobal().getXmlPostRequestLogBufferSize()
+                            : REQUEST_LOG_BUFFER_SIZE_DEFAULT; // Grabbed from global directly, not
+            // metadatamap for backwards
+            // compatibility
```
[the remark this answers is not in the document]
Comment (turingtestfail) [reply src/main/src/main/java/org/geoserver/filters/LoggingFilter.java]: fixed

File: src/main/src/main/java/org/geoserver/filters/LoggingFilter.java [hunk 4]
```diff
@@ -128,6 +142,7 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
                 }
 
                 if (logBodies
+                        && requestLogBufferSize > 0
```
[the remark this answers is not in the document]
Comment (turingtestfail) [reply src/main/src/main/java/org/geoserver/filters/LoggingFilter.java:144]: fixed

File: src/main/src/test/java/org/geoserver/filters/LoggingFilterTest.java [hunk 5]
```diff
@@ -52,23 +57,43 @@ public void setup() {
 
     @Test
     public void testRequestLoggingDoesNotOccur() throws IOException, ServletException {
-        String capturedLog = getLog("false", "true", "true");
+        String capturedLog =
+                getLog("false", "true", "true", LoggingFilter.REQUEST_LOG_BUFFER_SIZE_DEFAULT);
```
[the remark this answers is not in the document]
Comment (turingtestfail) [reply src/main/src/test/java/org/geoserver/filters/LoggingFilterTest.java]: fixed

**Review by aaime — APPROVED** [review]

**Pull request merged**